cyprecice/cyprecice.pyx

```diff
@@ -1211,6 +1211,249 @@ cdef class Interface:
+
+        Notes
+        -----
+        Previous calls:
+            Count of available elements at values matches the configured dimension
+            Count of available elements at vertex_ids matches the given size
+            Initialize() has been called
+            Data with dataID has attribute hasGradient = true
+            
+        Examples
+        --------
+        Write block gradient vector data for a 2D problem with 2 vertices:
+        >>> data_id = 1
+        >>> vertex_ids = [1, 2]
+        >>> gradientValues = np.array([[v1_dx, v1_dy],[v2_dx, v2_dy]])
+        >>> interface.write_block_vector_data(data_id, vertex_ids, gradientValues)
+
+        Write block vector data for a 3D (D=3) problem with 2 (N=2) vertices:
+        >>> data_id = 1
+        >>> vertex_ids = [1, 2]
+        >>> values = np.array([[v1_dx, v1_dy, v1x_dz], [v2_dx. v2_dy, v2_dz]])
+        >>> interface.write_block_scalar_data(data_id, vertex_ids, values)
+        """
+        check_array_like(vertex_ids, "vertex_ids", "write_block_scalar_gradient_data")
+        check_array_like(gradientValues, "gradientValues", "write_block_sclar_gradient_data")
+
+        if not isinstance(gradientValues, np.ndarray):
+            gradientValues = np.asarray(gradientValues)
+
+        if len(gradientValues) > 0:
+            size, dimensions = gradientValues.shape
+            assert dimensions == self.get_dimensions() , "Dimensions of vector data in write_block_scalar_gradient_data does not match with dimensions in problem definition. Provided dimensions: {}, expected dimensions: {}".format(dimensions, self.get_dimensions())
+        if len(gradientValues) == 0:
+            size = 0
+
+        cdef np.ndarray[int, ndim=1] _vertex_ids = np.ascontiguousarray(vertex_ids, dtype=np.int32)
+        cdef np.ndarray[double, ndim=1] _gradientValues = np.ascontiguousarray(gradientValues.flatten(), dtype=np.double)
+
+        assert _gradientValues.size == size * self.get_dimensions(), "Vector data is not provided for all vertices in write_block_vector_data. Check length of input data provided. Provided size: {}, expected size: {}".format(_gradientValues.size, size * self.get_dimensions())
+        assert _vertex_ids.size == size, "Vertex IDs are of incorrect length in write_block_scalar_gradient_data. Check length of vertex ids input. Provided size: {}, expected size: {}".format(_vertex_ids.size, size)
+
+        self.thisptr.writeBlockScalarGradientData (data_id, size, <const int*>_vertex_ids.data, <const double*>_gradientValues.data)
+
+    def is_gradient_data_required(self,data_id):
+        """
+        Returns true if gradient data is required for a dataID.
+        Parameters
+        ----------
+        data_id : int
+            Data ID to check.
+
+        Returns
+        -------
+        bool
+            True if gradient data is required for a dataID.
+
+        Notes
+        -----
+        Previous calls:
+            initialize() has been called
```

Comment:
That's not a requirement from preCICE. Is it one for python?

Reply:
I suspect this is just a copy paste from other functions. @kursatyurt please make sure that the required previous calls are really required for the new API.

Reply:
@DavidSCN I took it from, https://github.com/precice/precice/blob/d83d3c789172acca414233bd65524297c4ce3438/src/precice/SolverInterface.hpp#L1142 

Can we really call it before the initialization of preCICE?

Reply:
In doxygen, the documentation appears above the function declaration. In python, the documentation appears below the function declaration (right?). The `isRequired` function does not require `initialize`. The `writeGradient` functions require `initialize`.

Reply:
All clear, thank you.